src/frontends/tensorflow/src/op/if.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "input_model.hpp"
+#include "op_table.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::opset10;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_if_op(const NodeContext& node) {
+    default_op_checks(node, 1, {"If", "StatelessIf"});
+    auto node_name = node.get_name();
+    auto translate_session = node.get_translate_session();
+    FRONT_END_GENERAL_CHECK(translate_session, "[TensorFlow Frontend] Internal error: Translate session is nullptr.");
+
+    // retrieve body ov::Model for then and else branches
+    auto then_branch_type = node.get_attribute<std::string>("then_branch");
+    auto else_branch_type = node.get_attribute<std::string>("else_branch");
+    auto then_branch_body = translate_session->get_body_ov_model(then_branch_type);
+    FRONT_END_GENERAL_CHECK(
+        then_branch_body,
+        "[TensorFlow Frontend] Internal error or incorrect input model. Cannot find branch body graph with name " +
+            then_branch_type);
+    auto else_branch_body = translate_session->get_body_ov_model(else_branch_type);
+    FRONT_END_GENERAL_CHECK(
+        else_branch_body,
+        "[TensorFlow Frontend] Internal error or incorrect input model. Cannot find branch body graph with name " +
+            else_branch_type);
+
+    // get condition output
+    auto cond = node.get_input(0);
+    size_t input_size_t = node.get_input_size() - 1;
+    auto then_params = then_branch_body->get_parameters();
+    auto else_params = else_branch_body->get_parameters();
+    FRONT_END_GENERAL_CHECK(input_size_t == then_params.size(),
```

Comment:
Is it possible for tf If body to have less inputs then another body?

Reply:
No, it is not according to TF spec.